sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -270,6 +291,11 @@ case class CreateDataSourceTableAsSelectCommand(
   }
 }
 
+case class SchemaType private(name: String)
+object SchemaType {
```

Comment:
will we have more schema type? If not, I think a boolean flag `isSchemaInferred` should be good